### PR TITLE
EOS-27573: Added fix for busybox and changed master to primary.

### PR DESCRIPTION
**Pull request by nitisdev**
Signed-off-by: Nitish Singh <nitish.singh@seagate.com>

# Problem Statement
- By default during CORTX deployment on fresh VMs busybox image is being downloaded from docker hub even though passing the ghcr.io repo, which runs into errors due to rate limit thus halting deployement.

# Design
-  Added fix for busybox. local-path-provisioner is now deployed with CORTX deployment only.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- Fresh VMs single node: http://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/test_deploy_k8s_cortx_cluster/60/
- Fresh VMs 3 node: http://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/test_deploy_k8s_cortx_cluster/61/

After master change
- deploy-k8s-cortx-cluster: http://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/test_deploy_k8s_cortx_cluster/62/console 
- test-cortx-setup: http://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/nitisdev/job/test_cortx_setup/207
- prvsnr deployment: http://eos-jenkins.colo.seagate.com/job/Provisioner/job/prvsnr_framework_deployment/18/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by gauravchaudhari02 — DISMISSED** [review]

File: solutions/kubernetes/cortx-deploy-functions.sh [hunk 1]
```diff
@@ -256,8 +256,8 @@ fi
 
 function setup_master_node(){
 echo "---------------------------------------[ Setting up Master Node $HOSTNAME ]--------------------------------------"
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/cortx-deploy-functions.sh:258]: Please change Master to Primary

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
@nitisdev - Can you point me to the change for pulling busybox image.

**Comment by nitisdev** [comment]
> @nitisdev - Can you point me to the change for pulling busybox image.

Removed local-path-provisioner deployment from k8s setup step. Now its deployment only with CORTX.

File: solutions/kubernetes/cortx-deploy-functions.sh (hunk 1 above)
Comment (nitisdev) [reply solutions/kubernetes/cortx-deploy-functions.sh]: Done

**Review by gauravchaudhari02 — DISMISSED** [review]

**Comment by nitisdev** [comment]
> @nitisdev - Can you point me to the change for pulling busybox image.

Hence, no need to pull busybox explicitly now.

File: solutions/kubernetes/cortx-deploy.sh [hunk 2]
```diff
@@ -79,58 +79,58 @@ function setup_cluster {
...
        NODES=$(wc -l < $HOST_FILE)
-       TAINTED_NODES=$(ssh -o 'StrictHostKeyChecking=no' "$MASTER_NODE" bash << EOF
+       TAINTED_NODES=$(ssh -o 'StrictHostKeyChecking=no' "$PRIMARY_NODE" bash << EOF
 kubectl get nodes -o jsonpath="{range .items[*]}{.metadata.name} {.spec.taints[?(@.effect=='NoSchedule')].effect}{\"\n\"}{end}" | grep  NoSchedule | wc -l
 EOF
 )
        NODES="$((NODES-TAINTED_NODES))"
        echo "---------------------------------------[ $NODES node deployment ]----------------------------------"
-       echo "MASTER NODE:" $MASTER_NODE
+       echo "PRIMARY NODE:" $PRIMARYPRIMARY_NODE
```
Comment (shailesh-vaidya) [inline solutions/kubernetes/cortx-deploy.sh:97]: Should this bs only `$PRIMARY_NODE`

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: solutions/kubernetes/cortx-deploy.sh (hunk 2 above)
Comment (nitisdev) [reply solutions/kubernetes/cortx-deploy.sh]: Done

**Review by gauravchaudhari02 — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**